types/signatures.go

```diff
@@ -407,17 +417,72 @@ func (uh UnlockHash) MarshalJSON() ([]byte, error) {
 // that has been encoded to a hex string.
 func (uh *UnlockHash) UnmarshalJSON(b []byte) error {
 	// Check the length of b.
-	if len(b) != crypto.HashSize*2+2 {
+	if len(b) != crypto.HashSize*2+UnlockHashChecksumSize*2+2 {
+		return ErrUnlockHashWrongLen
+	}
+
+	// Decode the unlock hash.
+	var byteUnlockHash []byte
+	var checksum []byte
+	_, err := fmt.Sscanf(string(b[1:1+crypto.HashSize*2]), "%x", &byteUnlockHash)
+	if err != nil {
+		return err
+	}
+	// Decode the checksum.
+	_, err = fmt.Sscanf(string(b[1+crypto.HashSize*2:1+crypto.HashSize*2+UnlockHashChecksumSize*2]), "%x", &checksum)
+	if err != nil {
+		return err
+	}
```

Comment:
probably cleaner to split them up beforehand:
```go
// remove ""
b = b[1:len(b)-2]
// split unlock hash and checksum
uhBytes := b[:crypto.HashSize*2]
chkBytes := b[crypto.HashSize*2:]
// Decode...
```
It's safe to use a trailing `:` because the length has already been checked at the top of the function.